metabase/metabase
# use DateAllOptionsWidget for single date and range date filters

Closes https://github.com/metabase/metabase/issues/4585
Based on https://github.com/metabase/metabase/pull/18971

Replace single/range date dashboard filters with `DateAllOptionsWidget` filter for better UX:

#### Before

**Single date filter**
<img width="313" alt="Screen Shot 2021-12-14 at 20 00 56" src="https://user-images.githubusercontent.com/14301985/146045288-6b64ed7b-0381-46b2-80aa-e4ea97139217.png">

**Range date filter**
<img width="322" alt="Screen Shot 2021-12-14 at 20 01 01" src="https://user-images.githubusercontent.com/14301985/146045372-5918db04-c57f-4d67-9a6d-bc259bda98f0.png">

#### After

**Single date filter**
<img width="395" alt="Screen Shot 2021-12-14 at 19 59 00" src="https://user-images.githubusercontent.com/14301985/146045408-f05b8904-6877-47d7-b6de-f5cade50fc44.png">

**Range date filter**
<img width="562" alt="Screen Shot 2021-12-14 at 19 58 50" src="https://user-images.githubusercontent.com/14301985/146045449-c3d8d531-78af-40d3-88d9-ed8def3c2112.png">

### How to verify
- Create a dashboard with a question that has date columns
- Add single date and range date filters, save the dashboard
- Ensure the filters work as expected
